hack/tests/upgrade-e2e.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash -ex
+
+### Script flow:
+# 1. Source credentials from secret
+# 2. Create temporary folder for "source cluster" code
+# 3. Start monitoring shim layer from master
+#    We pass config file from temporary folder, 
+#    because it will be created there first during source cluster creation
+# 4. Link secrets to appropriate location
+# 5. Clone released code and create a cluster
+# 6. Switch to PR code base and inicate upgrade.
+#    Old cluster upgarde is always managed by new or same version of RP
+
+source_secret(){
+    # running in ci
+    if [[ -f /usr/local/e2e-secrets/azure/secret ]];then
```

Comment:
@mjudeikis follow-up: I'd rather that CI set up the symlink - our scripts in repo can/should expect secrets/secret to exist.